### ISSUE TITLE
GameGuru MAX - Bug? - Importer Changes Name if Name has Underscore

### ISSUE THREAD

**Issue by MonkeyFrogStudio** [TheGameCreators/GameGuruRepo#3676]
Currently, when you import a model, if the model's name has an underscore in it, MAX removes these and replaces them with a space. So, if your model is named:

SF_floor_red_01

You will get the following files:

SF floor red 01.BMP
SF floor red 01.dbo
SF floor red 01.fpe

The material names are not changed that I can see, so if they contain any underscores, they remain.

Normally, this is not really an issue. However, I'd prefer if MAX didn't alter my naming convention for models, if possible. One reason is I tend to name materials after the model. So, SF_floor_red_01.fbx would have material names like SF_floor_red_01_M. Because MAX removes the underscore for the DBO and FPE files, they end up separated from the materials when you look in the folder. This can make it more difficult to find all the parts and pieces if you want to delete them all or move them to a new folder, etc.

**Comment by ZakJudges** [comment]
Hi @MonkeyFrogStudio I have uploaded a new Steam experimental that fixes the issue of underscores being wiped out for imported objects

**Comment by MonkeyFrogStudio** [comment]
@ZakJudges Awesome! I will test today and get back to you.

**Comment by MonkeyFrogStudio** [comment]
@ZakJudges Okay, we seem to be MOSTLY there, but not all the way. I imported several models and the error was fixed. However, with this one particular object, I am still not getting the names of the materials to come in with it on import. Here's a screen shot:

![names](https://user-images.githubusercontent.com/36735507/201145522-5b0fd7dc-8ebf-412d-8648-120f89036cbd.png)

I am going to take another look at the model in Blender and see if there is anything I am doing wrong. If I find something, I will report back here. But so far this fix is not working on this object.

BTW - The object is made up of four models, each with their own material, exported from Blender as one FBX and then imported into MAX.

**Comment by MonkeyFrogStudio** [comment]
@ZakJudges - I just checked everything in Blender and can't see any errors. I deleted all reference to the model in my entitybank folder and re-imported only to get the same thing ... the material names aren't there, only the first name, "MFD" for all materials.

I can send you the model, but can't post it here as it's not for the public ... if that would help you.

**Comment by MonkeyFrogStudio** [comment]
@ZakJudges I've sent you the model and materials via email. If you don't get them, please let me know. I can send via another email address and I can also upload the files to my site if they are too large for you to get via email.

**Comment by ZakJudges** [comment]
@MonkeyFrogStudio I've downloaded the files. Thanks again

**Comment by MonkeyFrogStudio** [comment]
@ZakJudges Sorry, just trying to be clear - you mean you've gotten the email, because the file is attached. ;)

**Comment by ZakJudges** [comment]
@MonkeyFrogStudio I have uploaded another experimental with a possible fix. I haven't properly tested it yet, but if there are any more issues just let me know and I will take a look at it tomorrow :)

**Comment by MonkeyFrogStudio** [comment]
@ZakJudges Looks like this is about 98% fixed. Still a (very) minor issue, though. When the model comes in, the dropdown area for the material names still shows a truncated name:

![name1](https://user-images.githubusercontent.com/36735507/201170133-9878e6d4-73cb-4fcc-939a-64324552ad4c.png)

Sometimes, when you select a material in the list, it displays the correct name at the top, but sometimes it doesn't:

![name2](https://user-images.githubusercontent.com/36735507/201170249-02471e83-d6a1-4af8-897e-7826c47125e2.png)

Even so, it is possible, now, for people to see the correct names and select the material they are looking for.

**Comment by ZakJudges** [comment]
@MonkeyFrogStudio The emissive texture issue will take some more time but I have fixed the issue of the dropdown selection still having the wrong name. A new Steam experimental build is up with the fix.

**Comment by ZakJudges** [comment]
@MonkeyFrogStudio A new experimental build is up with the fix for the emissive issue. 

I also noticed that the previous fix for the material names were all offset by 1, due to the displaying of the root node in the list. This should also be fixed, but let me know if you encounter any more problems and I will look into it :)

**Comment by MonkeyFrogStudio** [comment]
@ZakJudges This all appears to be fixed! Thank you so much! I will close this one.

**Comment by MonkeyFrogStudio** [comment]
@ZakJudges 

I've reopened this because there appears to still be an issue with material names. I'd created individual floor parts, wall, parts, and ceiling parts and imported them in as individual pieces. Then I imported them together as one model. In the past, I could do this and there were no issues. Now, with the latest experimental build, it's pulling the model names for the material names (and only appears to be doing this when importing multiple models as one - like when you select several models and export them as one model). Here's a screen shot:

![names](https://user-images.githubusercontent.com/36735507/203156270-0a526355-d85a-4211-a1d7-cc1b8bdd5117.png)

The names in the material list in this case are all names of the models from in Blender and are not the material names that were associated with each model part.
